### ISSUE TITLE
I don't have CSS; should I be creating a local exclusion with XS71ECU1 in it?

### ISSUE THREAD

**Issue by benze** [dalgibbard/citrix_xenserver_patcher#73]
I just recently upgraded to XS 7.1 from an old version of XS6 and trying to set up automated patching to stay up to date.  From what I understand, I need a CSS subscription to be able to install XS71ECU1, or else, need to ensure I patch at most every quarter.

At the moment, patcher.py keeps failing every time it tries to install XS71ECU1 with an ```errorcode="LICENCE_RESTRICTION"```.

Should I simply be creating a local exclusion list and include XS71ECU1 in that list?  Or is the patcher.py smart enough to continue without that update and keep the server up to date with any future patches that are released?  

Can the patcher.py be updated to be smart enough that if CSS username/pwd credentials are not entered that it automatically skips the updates that require CSS?

Thanks,

Eric

**Comment by mario-filho-movile** [comment]
You need a subscription. You can configure a local exclusion. This
cimulative patch sets xs with version 7.1.1, so later hotfix will not work
with version 7.1

Em 29 de out de 2017 4:12 PM, "benze" <notifications@github.com> escreveu:

**Comment by dalgibbard** [comment]
If the existing patching pain wasn't enough to make people move platforms, the new licensing should be :(

**Comment by benze** [comment]
@dalgibbard agreed, but not sure what the options are.  Esx is no better, and MS is not a real option.  I bought into XS back at XS 4 or 5 (dont remember which) when the free model had real value.  But these days, it looks like they've pretty much shifted their entire model to the paid subscription.

I understand the reasoning; their pricing tiers though don't make sense to me.  Although there is still a free model, it kids being whittled down to nothing.

I thought I could potentially upgrade a standalone in my lab to 7.2 live via cli, but it looks like the only way to do that is to do it offline again.  Which is a nuisance only to keep up to date with patches.

Any suggestions what other solutions one can look at?

**Comment by dalgibbard** [comment]
It's an interesting question, and somewhat a matter of opinion really. We moved away from Xenserver pretty early on to VMware just because the feature set is pretty substantial in comparison; but probably so is the licensing fees, and as an open source enthusiast, it's hard to agree with either.

The choice largely depends on your own requirements, but Xen is always an option if virtualization is a fixed requirement; it's the basis of the commercial Citrix offering after all.

Otherwise, I'd highly recommend switching to a containerisation platform (docker), with the market leader being kubernetes. Mesos and Marathon (also packaged as DC/OS) is a decent alternative too imho, and things like CoreOS's automatic patching is an attractive option. The hardest part of a containerised platform is persistent storage, but there are a few viable options- I'd recommend shopping around for what meets your requirements, particularly as this market segment is still developing at a fairly rapid pace. We tended to keep container persistant storage to a minimum (stateless apps), and run things like databases as bare metal where performance was key.
